### ISSUE TITLE
Cache param will be a object

### ISSUE THREAD

**Issue by sanchis** [AlexMayol/localit#25]
Thinking in the future, it would be nice if the third parameter was optionally an object where you could add optional properties as the project grows.

This way we would greatly reduce the breaking changes generated by adding optional parameters.

Example: 
```js
store.set('i love the key', 2, { expiration: '2d' })
```

**Comment by AlexMayol** [comment]
What other properties or options can you think of?
```js
store.set('i love the key', 2, { 
expiration: '2d',
domain: 'domain',
})

```

**Comment by sanchis** [comment]
> What other properties or options can you think of?
> 
> ```js
> store.set('i love the key', 2, { 
> expiration: '2d',
> domain: 'domain',
> })
> ```
All these options should be optional.
```js
{
    expiration: '2ds',
    storage: 'localStorage', // localStorage or sessionStorage
    onExpire: (key, val) => { },
    onDelete: (key, val) => { },
    onUpdate: (key, val) => { },
    domain: 'SuperDomain'
}
```

**Comment by AlexMayol** [comment]
When should those functions be invoked? Keep in mind that `Storage` is not a good place to store functions, so they would be wiped out on a page refresh if we attach them to localit's instance.
I like the `storage` setting tho 👍

**Comment by sanchis** [comment]
> When should those functions be invoked? Keep in mind that `Storage` is not a good place to store functions, so they would be wiped out on a page refresh if we attach them to localit's instance. I like the `storage` setting tho 👍

You are right about the on* events, maybe we could generate them like this. 
```js
localit.onDelete('i love the key', (key,vale) => { })
localit.onUpdate('i love the key', (key,vale) => { }) 
localit.onExpire('i love the key', (key,vale) => { }) 
```
**But we'll see about that in another issue specifically for that.**

For the object options do you think it is ok to do it like this?
```js
{
    expiration: '2d',
    storage: 'localStorage', // localStorage or sessionStorage
    domain: 'SuperDomain'
}
```

**Comment by AlexMayol** [comment]
Yes, it looks great. 
I'm not sure if we should use that object as third parameter or also allow the current date string, because it's shorter to write and probably you won't change the Storage type and domain that frequently.

```js
// Ideally, both ways are allowed

store.set('key', value, '3h')

store.set('key', value, {expiration:'3h'})
```

**Comment by AlexMayol** [comment]
This improvement was shipped in [version 6](https://github.com/AlexMayol/localit/pull/58), thanks for the suggestion 🚀
